UndertaleModLib/Compiler/AssemblyWriter.cs

```diff
@@ -1376,9 +1376,18 @@ private static void AssembleExpression(CodeWriter cw, Parser.Statement e, Parser
                             cw.typeStack.Push(DataType.Variable);
                             cw.Emit(Opcode.Dup, DataType.Variable).Extra = 0;
                             if (isStructDef)
+                            {
                                 cw.Emit(Opcode.PushI, DataType.Int16).Value = (short)-16;
+                            }
+                            else if (cw.compileContext.OriginalCode?.Name?.Content == ("gml_GlobalScript_" + funcDefName.Text))
```

Comment:
Seems kinda messy but also dont know any other cleaner way.
I guess linq maybe, but dunno if we have access to the functions here?